components/profile-controller/pkg/controller/profile/profile_controller.go

```diff
@@ -249,16 +249,30 @@ func (r *ReconcileProfile) Reconcile(request reconcile.Request) (reconcile.Resul
 			"defaultEdittor", "error", err)
 		return reconcile.Result{}, err
 	}
+	gcpPlugin := &GcpPlugin{}
```

Comment:
I believe that Plugins should be called in an encapsulated way with a common entrypoint interface.
I wouldn't expect this common code to know about individual plugins.

Reply:
updated